Jenkins: disable ccache for clang

mrks opened the pull request:
Hopefully this stops CI from ruining everyone's day.

Seems to be a known issue: https://gitlab.kitware.com/cmake/cmake/issues/19923

Also removes the no-numa builds. We don't do too much differently there and the no-libnuma support is tested in the mac build anyway.

Bouncner commented:
Just curious: why disabling ccache and not PCH?

mrks commented:
You can't disable PCH from the outside, so we would need another CMake flag (and thus, a different configuration).

Bouncner commented on Jenkinsfile:
```diff
@@ -74,26 +74,26 @@ try {
           disable_aslr = 'setarch x86_64 -R'
 
           // Run cmake once in isolation and build jemalloc to avoid race conditions with autoconf (#1413)
-          sh "mkdir clang-debug && cd clang-debug &&                                                       ${cmake} ${ccache} ${debug}   ${clang}          .. && make -j libjemalloc-build"
```
Why only for debug? I don’t get what we are doing here from the comment.

mrks replied:
better?

Bouncner replied:
So it’s not about debug. It’s about running autoconf “anyhow” and afterwards everything works?

mrks replied:
correct

Bouncner replied:
Remove the empty line + replace “really” with “using debug here”)?

For my understanding: for cmake you need a new cmake run every time you change from debug to release. For automake, that’s not the case?

mrks replied:
Correct. I don't want to mention debug here because it has nothing to do with debug.

If you don't insist, I#d rather merge this once it's green. This is *the* PR needed to unfuck CI.

Bouncner replied:
The empty line still bothers me though. :D

Bouncner commented on Jenkinsfile:
```diff
@@ -74,26 +74,26 @@ try {
           // Configure the rest in parallel
-          // Most builds don't use unity builds as this would break the ccache use across different PRs. For clang-tidy, ccache does not have an effect, so we use unity there.
+          // clang + precompiled headers + ccache is broken: https://gitlab.kitware.com/cmake/cmake/issues/19923
+          // Some confirgurations don't use unity builds as this would break the ccache use across different PRs.
```
Typo

Bouncner commented on Jenkinsfile:
```diff
@@ -74,26 +74,26 @@ try {
 
         parallel clangDebug: {
           stage("clang-debug") {
-            sh "export CCACHE_BASEDIR=`pwd`; cd clang-debug && make all -j \$(( \$(cat /proc/cpuinfo | grep processor | wc -l) / 4)) && ../scripts/analyze_ccache_usage.py"
```
Aren’t you afraid we won’t put the analayze script back in once the cmake bug is fixed?

mrks replied:
Doesn't really matter. That was just for me to check things in the output of Jenkins. If we put ccache back in and forget it, the worst thing that could happen is someone missing and adding it.

Bouncner commented:
Just a note: depending on how long it takes to fix this issue, we might want to fix it outside of Jenkins as well. Got the same problem on nemea.

mrks commented:
Any chance you are changing files while a build is in progress? I saw the same thing when I started ninja and while waiting for the build to finished, changed the header again (e.g., added a comment).

Bouncner commented:
Nope. It was a benchmark run over night. After the first run (which built just fine), another branch was checked out and ninja was called. No concurrent modifications.

mrks commented:
Do you still have the error message? I'd like to see which files were affected.

Bouncner commented:
I am currently running a benchmark. After that, I can rebuild the tests which failed. Do you want the log or the folder to look into it yourself?

mrks commented:
Let's start with just one or two lines from the log.

Bouncner commented:
https://gist.github.com/Bouncner/af53fa445f66fb3401e62c3ede73f2c4

mrks commented:
Is that a nemea folder that I can access?

Bouncner commented:
Feel free. I am working in the `rel` directory.

mrks commented:
Working on the HP thing right now. If this is reproducible, let's look at it together next week.

Bouncner commented:
Ok.
